### ISSUE TITLE
🐛 [Bug] Can't use trt calibrator to calibrate SSD300

### ISSUE THREAD

**Issue by shiyongming** [NVIDIA/TRTorch#630]
## To Reproduce

1. install TensorRT 8.0.1.6
2. save 100 images into path `image-detection-dataset/data/`
3. run `python ssd300_traced_int8.py`

python ssd300_traced_int8.py is below and attached 
[ssd300_int8.zip](https://github.com/NVIDIA/TRTorch/files/7192464/ssd300_int8.zip)

```
import torch
torch.hub.list('NVIDIA/DeepLearningExamples:torchhub')

import torchvision
from torchvision import transforms

import trtorch

import tensorrt as trt
import os

import pycuda.driver as cuda
import pycuda.autoinit
from PIL import Image
import numpy as np

def data_Normalize(batch_data, mean_list=[0.4914, 0.4822, 0.4465], std_list=[0.2023, 0.1994, 0.2010]):
    batch_data[0, :, :] = (batch_data[0, :, :] - mean_list[0]) / std_list[0]
    batch_data[1, :, :] = (batch_data[1, :, :] - mean_list[1]) / std_list[1]
    batch_data[2, :, :] = (batch_data[2, :, :] - mean_list[2]) / std_list[2]
    return batch_data

# Returns a numpy buffer of shape (num_images, 3, 300, 300)
def load_for_ssd300(folder_path, total_images=10):
    image_idx = 0
    calib_images = np.zeros((total_images, 3, 300, 300))
    for filename in os.listdir(folder_path):
        print(folder_path + filename)
        img = Image.open(folder_path + filename)
        img = img.resize(size=(300, 300))        
        calib_images[image_idx] = np.array(img).transpose(2, 0, 1) / 255
        # calib_images[image_idx] = (calib_images[image_idx]-0.45)/0.20
        calib_images[image_idx] = data_Normalize(calib_images[image_idx])
        image_idx = image_idx + 1
    print('total_images:', total_images, 'calib_images shape:', calib_images.shape)
    return np.ascontiguousarray(calib_images.astype(np.float32))

class SSD300Calibrator(trt.IInt8EntropyCalibrator2):
    # class MNISTEntropyCalibrator(trt.IInt8MinMaxCalibrator):
    def __init__(self, calib_data_path, cache_file, images_num, batch_size=1):
        # Whenever you specify a custom constructor for a TensorRT class,
        # you MUST call the constructor of the parent explicitly.
        trt.IInt8EntropyCalibrator2.__init__(self)
        # trt.IInt8MinMaxCalibrator.__init__(self)

        # Prepare data for the the following
        self.data = load_for_ssd300(calib_data_path, images_num)
        self.cache_file = cache_file
        self.batch_size = batch_size
        self.current_index = 0

        # Allocate enough memory for a whole batch.
        self.device_input = cuda.mem_alloc(self.data[0].nbytes * self.batch_size)

    def get_batch_size(self):
        return self.batch_size

    def get_batch(self, names):
        if self.current_index + self.batch_size > self.data.shape[0]:
            return None

        # Print log for every 10 batches
        current_batch = int(self.current_index / self.batch_size)
        if current_batch % 10 == 0:
            print("Calibrating batch {:}, containing {:} images".format(current_batch, self.batch_size))

        # Every time get_batch is called, the next batch of size batch_size will be copied to the device and returned.
        batch = self.data[self.current_index:self.current_index + self.batch_size].ravel()
        cuda.memcpy_htod(self.device_input, batch)
        self.current_index += self.batch_size
        return [self.device_input]

    def read_calibration_cache(self):
        # If there is a cache, use it instead of calibrating again. Otherwise, implicitly return None.
        if os.path.exists(self.cache_file):
            with open(self.cache_file, "rb") as f:
                return f.read()

    def write_calibration_cache(self, cache):
        with open(self.cache_file, "wb") as f:
            f.write(cache)
            
precision = 'fp32'
ssd300 = torch.hub.load('NVIDIA/DeepLearningExamples:torchhub', 'nvidia_ssd', model_math=precision);

model = ssd300.eval().to("cuda")
traced_model = torch.jit.trace(model, [torch.randn((1,3,300,300)).to("cuda")])
traced_model.eval().to("cuda");

#print(testing_dataloader)
calib_data_path = 'image-detection-dataset/data/'
calibration_cache = "calibration.cache"
calibrator = SSD300Calibrator(calib_data_path, 
                              images_num=10, 
                              batch_size=1, 
                              cache_file=calibration_cache)
                              
batch_size = 1                                              
compile_spec = {
         "inputs": [trtorch.Input((batch_size, 3, 300, 300))],
         "enabled_precisions": {torch.float, torch.int8},
         "workspace_size": 1 << 20,
         "calibrator": calibrator,
         "device": {
             "device_type": trtorch.DeviceType.GPU,
             "gpu_id": 0,
#              "dla_core": 0,
#              "allow_gpu_fallback": False,
             "disable_tf32": False
         }
     }
trt_model_int8 = trtorch.compile(traced_model, compile_spec)
```

I got the error like below.
Traceback (most recent call last):
  File "ssd300_traced_int8.py", line 148, in <module>
    trt_model_int8 = trtorch.compile(traced_model, compile_spec)
  File "/opt/conda/lib/python3.8/site-packages/trtorch/_compiler.py", line 81, in compile
    compiled_cpp_mod = trtorch._C.compile_graph(module._c, _parse_compile_spec(compile_spec))
  File "/opt/conda/lib/python3.8/site-packages/trtorch/_compile_spec.py", line 204, in _parse_compile_spec
    info.ptq_calibrator = compile_spec["calibrator"]
TypeError: (): incompatible function arguments. The following argument types are supported:
    1. (self: trtorch._C.CompileSpec, arg0: trtorch._C.IInt8Calibrator) -> None

Invoked with: <trtorch._C.CompileSpec object at 0x7f4dae9c7c70>, <__main__.SSD300Calibrator object at 0x7f4dac0c21d0>

## Environment

> Build information about the TRTorch compiler can be found by turning on debug messages

 - TRTorch Version (e.g. 0.2.0): 0.4.0
 - PyTorch Version (e.g. 1.0): 1.9
 - CPU Architecture:
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source):
 - Build command you used (if compiling from source):
 - Are you using local sources or building from archives:
 - Python version:
 - CUDA version: 11.3
 - GPU models and configuration: T4
 - Any other relevant information:

**Comment by narendasan** [comment]
cc: @peri044

**Comment by peri044** [comment]
After looking into this, I suspect this might be due to pybind. To unblock you, you can replace the two instances of `trt.IInt8EntropyCalibrator2` with `trtorch._C.IInt8EntropyCalibrator2` in your code (with the rest of the calibrator being the same). 
After making this change, I hit a different error `RuntimeError: Method forward is not defined. ` The solution for this new issue is to use this PR : https://github.com/NVIDIA/TRTorch/pull/619 and rebuild the python API. We shall make a note of the first issue and investigate it further.

**Comment by shiyongming** [comment]
I have tried the dataloader calibrator. But got another issue. Code was uploaded.
For reproducing, please run `python ssd300_traced_int8_dataloader.py`
[ssd300_int8.zip](https://github.com/NVIDIA/TRTorch/files/7201297/ssd300_int8.zip)
.

**Comment by narendasan** [comment]
@peri044 what is TRT's current PyBind version?

**Comment by shiyongming** [comment]
> @peri044 what is TRT's current PyBind version?

Hi, how to check the TRT's Pybind version?

**Comment by peri044** [comment]
@shiyongming This bug is already fixed in PTQ calibrator provided in TRTorch. The fix is to pass empty string byte for read and write calibration cache functions. Please checkout this PR for details https://github.com/NVIDIA/TRTorch/pull/453
You need to modify your cache functions as follows 
```
def read_calibration_cache(self):
        # If there is a cache, use it instead of calibrating again. Otherwise, implicitly return None.
        if os.path.exists(self.cache_file):
            with open(self.cache_file, "rb") as f:
                return f.read()
        else:
            return b""

    def write_calibration_cache(self, cache):
        if self.cache_file:
            with open(self.cache_file, "wb") as f:
                f.write(cache)
        else:
            return b""
```

**Comment by shiyongming** [comment]
Thanks! I verified the above method. It works well now. Close the bug.
